etl/airflow.md

```diff
@@ -0,0 +1,329 @@
+functionality](https://luigi.readthedocs.io/en/stable/central_scheduler.html).
+2. Luigi is a build tool. [Its units of
+work](https://luigi.readthedocs.io/en/stable/tasks.html) are more Make-like in
+that they are tightly coupled to output from their composite targets.
+Conversely, Airflow operators run independently of one another, i.e., Airflow
+feels more conceptually appropriate for scheduling tasks that don’t necessarily
+create file-like artifacts, like scraping data.
+3. Airflow has eclipsed Luigi in
+popularity and maintenance resources, as measured by forks, stars, and watchers,
+and commits, respectively.
+
+Based on these impressions, and [an initial proof of
+concept](https://github.com/datamade/scrapers-airflow), we selected Airflow as
+our tool of choice for ETL management. Our first Airflow instance is the LA
+Metro Dashboard, which has been open sourced
+[here](https://github.com/datamade/la-metro-dashboard).
+
+## Basic concepts
+
+Airflow is a framework that relies on several key concepts. For a tutorial that
+guides you through a summary of these concepts in Airflow’s words, go to
+[https://airflow.apache.org/docs/stable/tutorial.html#example-pipeline-definition](https://airflow.apache.org/docs/stable/tutorial.html#example-pipeline-definition).
+Below are listed the most important Airflow concepts based on our work with
+Airflow thus far.
+
+### DAG (directed acyclic graph)
+
+A DAG is a way of visualizing the way Airflow works. It emphasizes that Airflow
+moves through tasks in a particular order without automatically repeating those
+tasks, and includes information about tasks’ dependencies on each other.
+
+- Airflow’s description:
+- [https://airflow.apache.org/docs/stable/concepts.html#dags](https://airflow.apache.org/docs/stable/concepts.html#dags)
+- A DAG run is an instance of a DAG.
+
+### Task
+
+A task is a single function or command to run. At the bare minimum,
+instantiating a task requires a `task_id`, `dag`, and the command argument for
+the particular operator being used.
+
+- For example, the `bash_operator` has a `command` argument, whereas the
+- `python_operator` has a `python_callable` argument. A task run is an instance
+- of a task.
+
+### Operator
+
+Airflow is really powerful in part because it can run commands for any part of
+your app that you can write in the most convenient language for that task [not
+sure about wording here]. For example, you can write a scraping script in Bash
+and run it with the `bash_operator`, run a Python command using the
+`python_operator`, and then run a database command using the `postgres_operator`
+-- all in the same DAG!
+
+There are also operators that handle control flow and other kinds of tasks. For
+a full list, see
+[https://airflow.apache.org/docs/stable/_api/airflow/operators/index.html](https://airflow.apache.org/docs/stable/_api/airflow/operators/index.html)
+
+### Pipeline
+
+A configuration file for a DAG. There are five parts to a pipeline, and they
```

Comment:
@fatima3558 I think it would be useful to put a link up top to Airflow's example! https://airflow.apache.org/docs/stable/tutorial.html#example-pipeline-definition